### PR TITLE
Duplicate fix

### PR DESCRIPTION
**Pull request by Cedricaca**
fix bug found by cheng zhen

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddCategoryCommandParser.java [hunk 1]
```diff
@@ -52,12 +56,15 @@ public AddCategoryCommand parse(String args) throws ParseException {
 
     }
 
-    /**
-     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
-     * {@code ArgumentMultimap}.
-     */
-    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
-        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    private boolean hasDuplicates(List<String> list) {
```
Comment (HongRay) [inline src/main/java/seedu/address/logic/parser/AddCategoryCommandParser.java:59]: Nice fix for duplicates. Will be using the same method for edit. Rmbr to add in assertion.

**Review by NusMinato — COMMENTED** [review]
Good job! Nice implementation to find the duplicate.

File: src/main/java/seedu/address/logic/parser/AddCategoryCommandParser.java [hunk 2]
```diff
@@ -52,12 +56,15 @@ public AddCategoryCommand parse(String args) throws ParseException {
...
+        Set<String> set = new HashSet<>();
+        for (String element : list) {
+            if (set.contains(element)) {
+                return true; // Found a duplicate
```
Comment (NusMinato) [inline src/main/java/seedu/address/logic/parser/AddCategoryCommandParser.java:63]: Nice implementation to find the duplicate.

**Review by NusMinato — COMMENTED** [review]
Good job! Nice implementation to find the duplicate.

File: src/main/java/seedu/address/logic/parser/AddCategoryCommandParser.java [hunk 3]
```diff
@@ -52,12 +56,15 @@ public AddCategoryCommand parse(String args) throws ParseException {
...
-    /**
-     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
-     * {@code ArgumentMultimap}.
-     */
-    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
-        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    private boolean hasDuplicates(List<String> list) {
+        Set<String> set = new HashSet<>();
+        for (String element : list) {
+            if (set.contains(element)) {
```
Comment (Xuyan0518) [inline src/main/java/seedu/address/logic/parser/AddCategoryCommandParser.java:62]: LGTM. Good job on implementing these duplicate checks.

**Pull request merged**